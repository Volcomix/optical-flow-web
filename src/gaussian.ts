const gaussian = (kernelSize: number, sigma = 0.15 * (kernelSize - 1)) => {
  const result: number[] = []
  const n = (kernelSize - 1) / 2
  for (let i = -n; i <= n; i++) {
    result.push(Math.exp((-0.5 * i ** 2) / sigma ** 2))
  }
  return result
}

export default gaussian
