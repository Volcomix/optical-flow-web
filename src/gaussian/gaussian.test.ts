import { expect, test } from 'vitest'

import gaussian from './gaussian'

test('gaussian', () => {
  expect(gaussian(11, 1.5)).toMatchSnapshot()
})

test('default sigma', () => {
  expect(gaussian(11)).toEqual(gaussian(11, 1.5))
})
