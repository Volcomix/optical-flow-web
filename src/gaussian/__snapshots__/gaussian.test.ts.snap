// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`gaussian 1`] = `
[
  0.0038659201394728076,
  0.028565500784550377,
  0.1353352832366127,
  0.41111229050718745,
  0.8007374029168081,
  1,
  0.8007374029168081,
  0.41111229050718745,
  0.1353352832366127,
  0.028565500784550377,
  0.0038659201394728076,
]
`;
